samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -36,56 +33,112 @@ static void sensor_cli_data_cb(struct bt_mesh_sensor_cli *cli,
+				       const struct bt_mesh_sensor_type *sensor, uint8_t index,
+				       uint8_t count,
+				       const struct bt_mesh_sensor_series_entry *entry)
 {
-	printk("Relative runtime in %d to %d degrees: %s percent\n",
-	       entry->value[1].val1, entry->value[2].val1,
-	       bt_mesh_sensor_ch_str(&entry->value[0]));
+	printk("Relative runtime in %d to %d degrees: %s percent\n", entry->value[1].val1,
+	       entry->value[2].val1, bt_mesh_sensor_ch_str(&entry->value[0]));
+}
+
+static void sensor_cli_setting_status_cb(struct bt_mesh_sensor_cli *cli,
+					 struct bt_mesh_msg_ctx *ctx,
+					 const struct bt_mesh_sensor_type *sensor,
+					 const struct bt_mesh_sensor_setting_status *setting)
+{
+	printk("Sensor ID: 0x%04x, ", sensor->id);
+	printk("Setting ID: 0x%04x\n", setting->type->id);
+	for (int chan = 0; chan < setting->type->channel_count; chan++) {
+		printk("Channel %d value: %s\n", chan,
+		       bt_mesh_sensor_ch_str(&(setting->value[chan])));
+	}
 }
 
 static const struct bt_mesh_sensor_cli_handlers bt_mesh_sensor_cli_handlers = {
 	.data = sensor_cli_data_cb,
 	.series_entry = sensor_cli_series_entry_cb,
+	.setting_status = sensor_cli_setting_status_cb,
 };
 
-static struct bt_mesh_sensor_cli sensor_cli =
-	BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
+static struct bt_mesh_sensor_cli sensor_cli = BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
 
 static struct k_work_delayable get_data_work;
 
 static void get_data(struct k_work *work)
 {
-	static bool req_temp_range;
+	static uint32_t sensor_idx;
 	int err;
 
 	/* Only one message can be published at a time. Swap sensor after each timeout. */
-	if (req_temp_range) {
+	switch ((sensor_idx++ % 3)) {
+	case (0): {
+		err = bt_mesh_sensor_cli_get(&sensor_cli, NULL, &bt_mesh_sensor_present_dev_op_temp,
+					     NULL);
+		if (err) {
+			printk("Error getting chip temperature (%d)\n", err);
+		}
+		break;
+	}
+	case (1): {
 		err = bt_mesh_sensor_cli_series_entries_get(
-			&sensor_cli, NULL,
-			&bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range, NULL, NULL,
-			NULL);
+			&sensor_cli, NULL, &bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range, NULL,
+			NULL, NULL);
 		if (err) {
-			printk("Error getting relative chip temperature data (%d)\n",
-			       err);
+			printk("Error getting relative chip temperature data (%d)\n", err);
 		}
-	} else {
-		err = bt_mesh_sensor_cli_get(
-			&sensor_cli, NULL, &bt_mesh_sensor_time_since_presence_detected,
-			NULL);
+		break;
+	}
+	case (2): {
+		err = bt_mesh_sensor_cli_get(&sensor_cli, NULL,
+					     &bt_mesh_sensor_time_since_presence_detected, NULL);
 		if (err) {
-			printk("Error getting time since presence detected (%d)\n",
-			       err);
+			printk("Error getting time since presence detected (%d)\n", err);
 		}
+		break;
+	}
 	}
+	k_work_schedule(&get_data_work, K_MSEC(GET_DATA_INTERVAL));
+}
 
-	req_temp_range = !req_temp_range;
+static uint32_t tidx;
```

Comment:
If this variable is only used inside the button handler, it should be decleared there. Also I would prefer if you changed the variable name to something else, like `temp_idx`. `tidx` can be a bit ambiguous since `tid` often will be read as "Transaction ID"